internal/alb/ls/listener.go

```diff
@@ -333,3 +338,62 @@ func (l *Listener) DefaultActionArn() *string {
 	}
 	return nil
 }
+
+func getCertificates(certificateArn *string, ingress *extensions.Ingress, logger *log.Logger) ([]*elbv2.Certificate, error) {
+	if certificateArn != nil {
+		logger.Debugf("New desired listener has certificate-arn '%v' in annotation", certificateArn)
+		return []*elbv2.Certificate{
+			{CertificateArn: certificateArn},
+		}, nil
+	}
+
+	logger.Debugf("New desired listener wants HTTPS, but hasn't provided an certificate-arn annotation")
+	certificates, err := albacm.ACMsvc.ListCertificates(&acm.ListCertificatesInput{
```

Comment:
Can we use the [paginated ListCertificates](https://docs.aws.amazon.com/sdk-for-go/api/service/acm/#ACM.ListCertificatesPages) instead?